### DISCUSSION TITLE
How to enable CORS with socketioxide used as a service (warp)

### DISCUSSION THREAD

**Discussion by Tryum** [Totodore/socketioxide#191] [Q&A]
CORS is working on the requests served by warp, but not the `/socket.io/` requests.

**Comment by Totodore** [comment]
I suppose you tried to use this https://docs.rs/warp/latest/warp/filters/cors/index.html.
Because Warp doesn't rely on tower service/layers, the filter is not applied to socket.io. Socket.io _wraps_ the warp instance and therefore all warp stuff is not apply to the socketioxide lib.

I will think about something to modify input/output requests so it is possible to apply CORS policies.

In the meantime you can still wrap the service fn with another service fn and manually modify headers to apply CORS policies : 
* https://docs.rs/hyper/0.14.27/hyper/server/index.html#examples
